commands/addrole.js

```diff
@@ -0,0 +1,127 @@
+    let channels = message.guild.channels
+    channels = channels.first(channels.array().length)
+    // channels is now an arrray of all the channels in this server
+    let l = channels.length
+    let i
+    // now keep looping through the text channels and try to fetch the message
+    for (i = 0; i < l; i++) {
+        if (channels[i].type == "text") {
+            good_message = await channels[i].fetchMessage(message_id).catch(err => { })
+            // once message is found, react to it with the given emoji
+            if (good_message) {
+                await good_message.react(emoji).catch(err => { })
+                channel_id = channels[i].id
+            }
+        }
+    }
+    // if no role was mentioned in the command call
+    if (!role) {
+        return message.channel.send(
+            clean
+                .setColor("#FF0000")
+                .addField("Error", "No roles were mentioned in your command, please mention the role to add.")
+        ).then(r => r.delete(30000))
+    }
+    // check if there's an emoji already bound to that message with a role.
+    reaction_message_model.findOne({
+        messageID: message_id
+    }, async (err, reaction_message) => {
+        msg = await message.reply("Loading...")
+        if (err) console.log(err)
+        // create a new object of the model to add it to the database
+        if (!reaction_message) {
+            const new_reaction_message = new reaction_message_model({
+                guildID: message.guild.id,
+                channelID: channel_id,
+                messageID: message_id,
+                roles: {}
+            })
+            new_reaction_message.roles[emoji] = {}
+            new_reaction_message.roles[emoji]["name"] = role.name
+            new_reaction_message.roles[emoji]["id"] = role.id
+            new_reaction_message.save().catch(err => console.log(err))
+        } else {
+            // check if the same emote we're trying to add already exists in the database
+            if (emoji in reaction_message.roles) {
+                const filter = m => m.author.id === message.author.id
+                // ask the user if they want to overwrite the current role that's bounded to this emoji in the database
+                let overwrite = new Discord.RichEmbed()
+                    .setColor("#FFA500")
+                    .addField("Warning", `The emote ${emoji} was already found in the database on the mentioned message for the role **${role.name}**, Would you like to overwrite it?`)
+                    .addField("Respone", "Type **Yes** to overwrite or anything else to Cancel")
+                    .setDescription("*This command will be automatically cancelled in 60 seconds*")
+                msg.edit(overwrite)
+                // await for the user's reply for 60 seconds.
+                var eternity = await message.channel.awaitMessages(filter, { max: 1, time: 60000 }).then(async collected => {
```

Comment:
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
                var eternity = await message.channel.awaitMessages(filter, { max: 1, time: 60000 }).then(async function(collected) {
```